backend/schema/metadataretry.go

```diff
@@ -0,0 +1,124 @@
+package schema
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/types/optional"
+	"google.golang.org/protobuf/proto"
+
+	schemapb "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/schema"
+)
+
+const (
```

Comment:
String constants are just to make error printing have the same format we expect users to use (ie `1d` not go's standard `24h0m0s`